tests/integration/test_refreshable_mat_view/test.py

```diff
@@ -0,0 +1,864 @@
+        refresh_interval=interval,
+        table_clause="ENGINE = Memory",
+        select_query="SELECT now() as a, number as b FROM numbers(2)",
+        with_append=append,
+        empty=empty,
+    )
+    node.query(create_sql)
+
+    now = datetime.utcnow()
+    rmv = get_rmv_info(node, "test_rmv")
+    predicted_next_refresh_time = get_next_refresh_time(interval, now, first_week=True)
+    compare_dates(
+        rmv["next_refresh_time"], predicted_next_refresh_time, first_week=True
+    )
+    assert rmv["next_refresh_time"] > now
+    assert rmv["exception"] is None
+
+    if empty:
+        # No data is inserted with empty
+        expect_rows(0)
+    else:
+        # Query is executed one time
+        compare_dates_(rmv["last_success_time"], now)
+        expect_rows(2)
+
+    # TODO: SET FAKE TIME doesn't work for these
+    if "RANDOMIZE" in interval:
+        return
+    if "OFFSET" in interval:
+        return
+
+    # Trigger refresh, update `next_refresh_time` and check interval again
+    node.query(f"SYSTEM TEST VIEW test_rmv SET FAKE TIME '{rmv['next_refresh_time']}'")
+    predicted_next_refresh_time2 = get_next_refresh_time(
+        interval, rmv["next_refresh_time"]
+    )
+    rmv2 = get_rmv_info(
+        node,
+        "test_rmv",
+        condition=lambda x: x["next_refresh_time"] != rmv["next_refresh_time"],
+    )
+    compare_dates(rmv2["next_refresh_time"], predicted_next_refresh_time2)
+    assert rmv["exception"] is None, rmv
+
+    if append and not empty:
+        expect_rows(4)
+    else:
+        expect_rows(2)
+
+    compare_dates(
+        rmv2["last_success_time"], predicted_next_refresh_time, first_week=True
+    )
+
+
+@pytest.fixture(scope="function")
+def fn2_setup_tables():
+    node.query(
+        f"CREATE TABLE tgt1 ON CLUSTER test_cluster (a DateTime, b UInt64) ENGINE = MergeTree ORDER BY tuple()"
+    )
+
+    node.query(
+        f"CREATE TABLE src1 ON CLUSTER test_cluster (a DateTime, b UInt64) ENGINE = Memory"
+    )
+    node.query(f"INSERT INTO src1 VALUES ('2020-01-01', 1), ('2020-01-02', 2)")
+
+    yield
+    node.query("DROP TABLE IF EXISTS src1 ON CLUSTER test_cluster")
+    node.query("DROP TABLE IF EXISTS tgt1 ON CLUSTER test_cluster")
+    node.query("DROP TABLE IF EXISTS test_rmv ON CLUSTER test_cluster")
+
+
+@pytest.mark.parametrize(
+    "append",
+    [True, False],
+)
+# @pytest.mark.parametrize("on_cluster", [True, False])
+@pytest.mark.parametrize(
+    "empty",
+    [True, False],
+)
+@pytest.mark.parametrize(
+    "to_clause",
+    [(None, "tgt1", "tgt1"), ("Engine MergeTree ORDER BY tuple()", None, "test_rmv")],
+)
+def test_real_wait_refresh(
+    fn2_setup_tables,
+    started_cluster,
+    append,
+    # on_cluster,
+    empty,
+    to_clause,
+):
+    table_clause, to_clause_, tgt = to_clause
+
+    create_sql = CREATE_RMV_TEMPLATE.render(
+        table_name="test_rmv",
+        # if_not_exists=if_not_exists,
+        refresh_interval="EVERY 8 SECOND",
```

Comment:
Can also test that the first refresh happens immediately, out-of-schedule (unless EMPTY). Use `refresh every 16 second offset {(now + 8 seconds) % 16 seconds}`, then check that the first 3 refreshes happen ~[0, 8, 24] seconds after view creation. (Maybe don't actually wait for the 24-second mark and just check `next_refresh_time`.)